server/aesdsocket.c

```diff
@@ -334,41 +338,56 @@ void *serve_client(void *thread_info)
                     goto exit_all;  
                 }
 #endif
-                int written_bytes;
-                int len_to_write = index;
-                char *ptr_to_write = recv_data;
-                while(len_to_write != 0)
+
+                if(strncmp(recv_data, IOCTL_CMD, IOCTL_CMD_LEN) == 0)
                 {
-                    written_bytes = write(file_fd, ptr_to_write, len_to_write);
-                    if(written_bytes == -1)
+                    struct aesd_seekto seekto;
+                    sscanf(recv_data, "AESDCHAR_IOCSEEKTO:%d,%d", &seekto.write_cmd, &seekto.write_cmd_offset);
```

Comment:
@jordicrosmompart you need error handling here for `sscanf` to handle invalid input